uber/site_sections/budget.py

```diff
@@ -46,3 +46,157 @@ def mpoints(self, session):
         all = [(sum(mpu.amount for mpu in mpus), group, mpus)
                for group, mpus in groups.items()]
         return {'all': sorted(all, reverse=True)}
+
+    @ajax
+    def add_promo_code_words(self, session, text='', part_of_speech=0):
```

Comment:
I also have reservations about promo code stubs. It's a neat feature, but unless we have a requirement for it coming from a specific use case, it's also a bunch of additional code surface that we might not have to add.